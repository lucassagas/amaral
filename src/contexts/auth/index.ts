export { AuthContext, AuthContextProvider, handleSignOut } from './auth'
