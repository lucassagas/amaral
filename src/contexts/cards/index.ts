export { CardContext, CardContextProvider } from './cards';

