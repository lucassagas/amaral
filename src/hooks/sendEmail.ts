import { api } from '@/services/api'
import { useCallback } from 'react'
// import {
//   Si
// } from '../contexts/cards/models'

const sendEmail = useCallback(async ({ data }: any): Promise<void> => {
  try {
    const { data: newData } = await api.post<any>('api/send',{
      data
    })

    console.log(newData)
  } catch(err){
    console.log(err)
  } finally {
    console.log('uepa')
  }
}, [])
