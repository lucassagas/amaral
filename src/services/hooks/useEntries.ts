import { formatDate } from '@/utils/formatDate'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

export type EntriesProps = {
  id: string
  oldQuantity: number
  newQuantity: number
  quantity: number
  updatedAt: Date | string
  product: {
    name: string
  }
  createdBy: {
    name: string
  }
}

type ApiResponse = {
  entries: EntriesProps[]
  totalEntries: number
}

async function getEntries(page: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/entries', {
    params: page,
  })

  const formatedEntries = data.entries.map((entry) => ({
    ...entry,
    updatedAt: formatDate(entry.updatedAt),
  }))

  const formatedData = {
    totalEntries: data.totalEntries,
    entries: formatedEntries,
  }

  return formatedData
}

export function useEntries(
  page: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['entries', page], () => getEntries(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
