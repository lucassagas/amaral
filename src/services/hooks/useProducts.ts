import { Product } from '@prisma/client'
import {
  QueryOptions,
  useQuery,
  UseQueryOptions,
  UseQueryResult,
} from 'react-query'
import { api } from '../api'

type ApiResponse = {
  products: Product[]
  totalProducts: number
}

async function getProducts(page?: number): Promise<ApiResponse> {
  const { data } = await api.get<ApiResponse>('api/products', {
    params: page,
  })

  return data
}

export function useProducts(
  page?: number,
  options?: UseQueryOptions<ApiResponse>
): UseQueryResult<ApiResponse> {
  return useQuery(['products', page], () => getProducts(page), {
    staleTime: 1000 * 60 * 10, // 10 minutes
    ...(options as QueryOptions),
  })
}
