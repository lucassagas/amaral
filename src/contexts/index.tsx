import { theme } from '@/styles/theme'
import { ChakraProvider } from '@chakra-ui/react'
import { ReactNode } from 'react'
import { QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import { queryClient } from 'src/lib/queryClient'
import { AuthContextProvider } from './auth'

type AppProviderProps = {
  children: ReactNode
}

export function AppProvider({ children }: AppProviderProps) {
  return (
    <ChakraProvider theme={theme}>
      <AuthContextProvider>
        <QueryClientProvider client={queryClient}>
          {children}
          <ReactQueryDevtools />
        </QueryClientProvider>
      </AuthContextProvider>
    </ChakraProvider>
  )
}
