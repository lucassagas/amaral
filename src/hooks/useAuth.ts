import { AuthContext } from '@/contexts/auth'
import { AuthContextData } from '@/contexts/auth/model'
import { useContext } from 'react'

export function useAuth(): AuthContextData {
  const context = useContext(AuthContext)

  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider')
  }

  return context
}
