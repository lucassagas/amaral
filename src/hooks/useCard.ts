import { CardContext } from '@/contexts/cards'
import { CardContextData } from '@/contexts/cards/models'
import { useContext } from 'react'

export function useCard(): CardContextData {
  const context = useContext(CardContext)

  if (!context) {
    throw new Error('useCard must be used within an CardProvider')
  }

  return context
}
