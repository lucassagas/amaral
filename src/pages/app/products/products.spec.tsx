import { AppProvider } from '@/contexts/index'
import { useProducts } from '@/services/hooks/useProducts'
import { fireEvent, render, screen } from '@testing-library/react'
import { act } from '@testing-library/react-hooks'
import { GetServerSidePropsContext } from 'next'
import Products, { getServerSideProps } from '.'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
})

jest.mock('@/services/hooks/useProducts', () => ({
  useProducts: jest.fn(),
}))

const products = [
  {
    id: '123',
    name: 'John Doe',
    quantity: 100,
    createdAt: new Date(),
    updatedAt: new Date(),
  },
]

const renderComponent = () => {
  return render(
    <AppProvider>
      <Products />
    </AppProvider>
  )
}

describe('<Products/>', () => {
  it('should render correctly', () => {
    ;(useProducts as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      data: {
        products: products,
      },
    }))

    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should render error when return error', () => {
    ;(useProducts as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      error: true,
    }))
    renderComponent()

    expect(
      screen.getByText(/falha ao obter dados dos produtos\./i)
    ).toBeInTheDocument()
  })

  it('should render loading when isLoading', () => {
    ;(useProducts as jest.Mock).mockImplementation(() => ({
      isLoading: true,
    }))
    renderComponent()

    expect(screen.getByText(/loading\.\.\./i)).toBeInTheDocument()
  })

  it('should refetch data when click in refetch button', async () => {
    ;(useProducts as jest.Mock).mockImplementation(() => ({
      isLoading: false,
      refetch: jest.fn(),
    }))

    renderComponent()

    const button = screen.getByRole('button')

    await act(async () => {
      fireEvent.click(button)
    })

    expect(useProducts).toHaveBeenCalled()
  })

  it('should test server side props ', async () => {
    const response = await getServerSideProps({} as GetServerSidePropsContext)

    expect(response).toBeDefined()
  })
})
