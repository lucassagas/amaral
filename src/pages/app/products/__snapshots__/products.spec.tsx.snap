// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products/> should render correctly 1`] = `
<div>
  <div
    class="css-1vfdvw2"
  >
    <div
      class="css-6iav15"
    >
      <h2
        class="chakra-heading css-sxvlx"
      >
        Cadastro de Produtos
      </h2>
      <div
        class="css-k008qs"
      >
        <button
          class="chakra-button css-fkz59o"
          type="button"
        >
          <svg
            class="chakra-icon css-12tbk7p"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5.463 4.433A9.961 9.961 0 0 1 12 2c5.523 0 10 4.477 10 10 0 2.136-.67 4.116-1.81 5.74L17 12h3A8 8 0 0 0 6.46 6.228l-.997-1.795zm13.074 15.134A9.961 9.961 0 0 1 12 22C6.477 22 2 17.523 2 12c0-2.136.67-4.116 1.81-5.74L7 12H4a8 8 0 0 0 13.54 5.772l.997 1.795z"
              />
            </g>
          </svg>
        </button>
        <a
          class="chakra-button css-1hvb8bx"
          href="/app/products/create"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-1cv0b1u"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M11 11V5h2v6h6v2h-6v6h-2v-6H5v-2z"
                />
              </g>
            </svg>
          </span>
          Cadastrar
        </a>
      </div>
    </div>
    <div
      class="css-1j4vx6g"
    >
      <table
        class="chakra-table css-e4kp03"
        role="table"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <th
              class="css-1ednkar"
            >
              Nome
            </th>
            <th
              class="css-1ednkar"
            >
              <p
                class="chakra-text css-xi606m"
              >
                Quantidade
              </p>
            </th>
            <th
              class="css-ai8wzg"
            />
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-0"
              >
                John Doe
              </p>
            </td>
            <td
              class="css-1n48h1c"
              role="gridcell"
            >
              <p
                class="chakra-text css-xi606m"
              >
                100
              </p>
            </td>
            <td
              class="css-1k2clil"
              role="gridcell"
            >
              <div
                class="chakra-stack css-84zodg"
              >
                <a
                  class="chakra-button css-vx0rs5"
                  href="/app/products/update/123"
                >
                  <span
                    class="chakra-button__icon css-1wh2kri"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon css-ky7im0"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g>
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M15.728 9.686l-1.414-1.414L5 17.586V19h1.414l9.314-9.314zm1.414-1.414l1.414-1.414-1.414-1.414-1.414 1.414 1.414 1.414zM7.242 21H3v-4.243L16.435 3.322a1 1 0 0 1 1.414 0l2.829 2.829a1 1 0 0 1 0 1.414L7.243 21z"
                        />
                      </g>
                    </svg>
                  </span>
                </a>
                <a
                  class="chakra-button css-160fvem"
                >
                  <span
                    class="chakra-button__icon css-1wh2kri"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon css-ky7im0"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15 2H9c-1.103 0-2 .897-2 2v2H3v2h2v12c0 1.103.897 2 2 2h10c1.103 0 2-.897 2-2V8h2V6h-4V4c0-1.103-.897-2-2-2zM9 4h6v2H9V4zm8 16H7V8h10v12z"
                      />
                    </svg>
                  </span>
                </a>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="chakra-stack css-1fjzvpp"
      >
        <div
          class="css-0"
        >
          <strong>
            1
          </strong>
           -
           
          <strong>
            10
          </strong>
           
          de 
          <strong />
        </div>
        <div
          class="chakra-stack css-wta9zy"
        >
          <button
            class="chakra-button css-1jw1j19"
            disabled=""
            type="button"
          >
            1
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
